starfish/imagestack/imagestack.py

```diff
@@ -847,44 +847,19 @@ def transform(
         if verbose and StarfishConfig().verbose:
             selectors_and_slice_lists = tqdm(selectors_and_slice_lists)
 
-        if n_processes == 1:
-            sp_applyfunc: Callable = partial(
-                self._singleprocessing_workflow,
-                partial(func, **kwargs),
-                self._data.data.values,
-            )
-
-            sp_results = []
-            for selector, slice_list in selectors_and_slice_lists:
-                result = sp_applyfunc((selector, slice_list))
-                sp_results.append((result, selector))
-            return sp_results
-        else:
+        with Pool(
+                processes=n_processes,
+                initializer=SharedMemory.initializer,
+                initargs=((self._data._backing_mp_array,
+                           self._data._data.shape,
+                           self._data._data.dtype),)) as pool:
             mp_applyfunc: Callable = partial(
-                self._multiprocessing_workflow, partial(func, **kwargs))
-
-            with multiprocessing.Pool(
-                    n_processes,
-                    initializer=SharedMemory.initializer,
-                    initargs=((self._data._backing_mp_array,
-                               self._data._data.shape,
-                               self._data._data.dtype),)) as pool:
-                mp_results = pool.imap(mp_applyfunc, selectors_and_slice_lists)
-                return list(zip(mp_results, selectors))
-
-    @staticmethod
-    def _singleprocessing_workflow(
-            worker_callable: Callable[[np.ndarray], Any],
-            numpy_array: np.ndarray,
-            selector_and_slice_list: Tuple[Mapping[Axes, int],
-                                           Tuple[Union[int, slice], ...]],
-    ):
-        sliced = numpy_array[selector_and_slice_list[1]]
-
-        return worker_callable(sliced)
+                self._processing_workflow, partial(func, **kwargs))
+            results = pool.imap(mp_applyfunc, selectors_and_slice_lists)
```

Comment:
I notice we're using imap here, but we only wrap map in the multiprocessing module. Is this a problem?

Reply:
ah good catch! It didn't matter with the two processing methods but now it does.